### PR TITLE
Associations & Authorization checking

### PR DESCRIPTION
**Pull request by knowtheory**
Had to back out the migrations, but now the data store is set up (and rails compatible) in such a way that we can access all authorizations and the most recent authorization by team.

Additionally this branch contains a demonstration of how to use Sequelize to do a check to see if a team is verified or not (in `slack-events-api-message-handler/index.understudy.js`).

### CODE REVIEW

File: slack-events-api-message-handler/index.understudy.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+var Sequelize = require('sequelize');
+var db        = require('./lib/models/db')(Sequelize);
+
+var responder = function(event, context, callback){
+    db.team.findOne({ where: { slack_id: event.team_id } })
+    .then( team => {
+        if (team === null) {
+            // bail.  We somehow got a message from a team
```
Comment (chriszarate) [inline slack-events-api-message-handler/index.understudy.js]: maybe log this?

File: slack-events-api-message-handler/index.understudy.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            
+            var bot_details      = authorization.details.bot;
+            var bot_access_token = bot_details.bot_access_token;
+            var bot_user_id      = bot_details.bot_user_id;
+            
+            // Tell the team they're not cool enough.
+            if (!team.verified) {
+                console.log('Team not yet validated by DocumentCloud. Informing user ...');
+                var message = "I'm still waiting for the folks at DocumentCloud to say you can use my services!";
+                //return sendToSlack(event, message);
```
Comment (chriszarate) [inline slack-events-api-message-handler/index.understudy.js]: uncomment?

File: slack-events-api-message-handler/index.understudy.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            } else {
+                // add the authorization info to the event
+                console.log('Team Verified, handling message');
+                event.authorization = authorization;
+                //return respondToSlackMessage( event, context, callback);
```
Comment (chriszarate) [inline slack-events-api-message-handler/index.understudy.js]: uncomment?

**Comment by chriszarate** [comment]
🎉

**Comment by knowtheory** [comment]
Okay!  I'm going to merge this and move to another PR to do the extraction of the TeamStore as a lib.

**Pull request merged**